### DISCUSSION TITLE
Where is the session data stored?

### DISCUSSION THREAD

**Discussion by dushyantbangal** [oidc-wp/openid-connect-generic#438] [Q&A]
Does the plugin manage sessions by itself? If yes, where is the session related data stored?

Initially I thought it might be using the php session management itself. But there are no files in `/tmp` directory after I login in my application. However I see files in the `/tmp` directory when I just added a php file which does `session_start()` and accessed that file's URL via my browser.

Are there any settings that I should be looking at for this?

**Comment by dushyantbangal** [comment]
The issue I am facing is that the user gets logged out when my kubernetes pod restarts or gets killed.

This happens even when I still have the wordpress_sec cookie in my browser, and the `wp_usermeta` still have the token in it. This makes me think something is being written in the pod itself.

**Comment by timnolte** [comment]
@dushyantbangal sorry for the delay in responding to you question. I've converted this to a Q&A Discussion topic.

To answer your question the plugin doesn't establish its own session but establishes a standard WordPress session. The only additional aspect however is that there are some states stored in the database, and there is a refresh token item that can cause a WordPress session to be logged out if the IDP puts a limit on the length of a logged in session.

**Reply by dushyantbangal** [reply]
@timnolte thanks for the reply.
You're saying it establishes a standard Wordpress session, and from what I see in wordpress docs, they establish a standard php session.

However, I dont see anything in the php session (file system based) directory. When I call the php create session function in a simple php file, I do get the php session. That led me to believe this plugin is handling it in some other way.

**Reply by timnolte** [reply]
FYI, the process for logging in a WP user, after IDP authentication, is here:

https://github.com/oidc-wp/openid-connect-generic/blob/99ec7c8cc743a194b6372500ded1dead02d08564/includes/openid-connect-generic-client-wrapper.php#L665-L684
